subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -40,62 +40,103 @@ import (
 // Note that the topologyAwareScheduler used in this struct is not another implementation of SchedulerAlgorithm;
 // that is a specific algorithm for pod placement, used in intra-VC scheduling and opportunistic pod scheduling.
 type HivedAlgorithm struct {
+	// data structures for the scheduling algorithm
+
 	// scheduler in each VC
 	vcSchedulers map[api.VirtualClusterName]intraVCScheduler
 	// scheduler for opportunistic pods
 	opportunisticSchedulers map[CellChain]*topologyAwareScheduler
-	// ChainCellLists of physical cells of each cell chain
+	// ChainCellLists of physical cells of each cell chain (including the children of the free cells)
 	fullCellList map[CellChain]ChainCellList
 	// ChainCellLists of free physical cells of each cell chain (used in buddy alloc)
 	freeCellList map[CellChain]ChainCellList
+	// all affinity groups that have been allocated cells
+	allocatedAffinityGroups map[string]*AlgoAffinityGroup
+
+	// The three cell numbers below are used to track cell usage of the VCs.
+	// Note that these numbers count both healthy and bad cells.
+
+	// number of free cells of each VC of each cell type
+	numVCFreeCells map[api.VirtualClusterName]map[CellChain]map[CellLevel]int32
```

Comment:
>numVCFreeCells [](start = 1, length = 14)

Prefer to move word "num" to tail, such as vcFreeCellNum, totalLeftCellNum, totalBadCellNum, etc.

This is to align with cell"Type"/"Num"/"Addr", instead of "type"Cell, etc